numba_dpex/tests/experimental/kernel_iface/spv_overloads/test_atomic_fetch_phi.py

```diff
@@ -14,32 +16,297 @@
     no_bool=True, no_float16=True, no_none=True, no_complex=True
 )
 
+list_of_int_dtypes = get_all_dtypes(
+    no_bool=True, no_float=True, no_none=True, no_complex=True
+)
+
+list_of_float_dtypes = get_all_dtypes(
+    no_bool=True, no_int=True, no_float16=True, no_none=True, no_complex=True
+)
+
 
 @pytest.fixture(params=list_of_supported_dtypes)
-def input_arrays(request):
+def input_arrays_for_add(request):
     # The size of input and out arrays to be used
     N = 10
     a = dpnp.ones(N, dtype=request.param)
     b = dpnp.zeros(N, dtype=request.param)
     return a, b
 
 
+@pytest.fixture(params=list_of_supported_dtypes)
+def input_arrays_for_sub(request):
+    # The size of input and out arrays to be used
+    N = 10
+    a = dpnp.ones(N, dtype=request.param)
+    b_np = np.ones(N) * N + 1
+    b = dpnp.asarray(b_np, dtype=request.param)
+    return a, b
+
+
+@pytest.fixture(params=list_of_supported_dtypes)
+def input_arrays_for_min_max(request):
+    # The size of input and out arrays to be used
+    N = 10
+    a = dpnp.arange(N, dtype=request.param)
+    b = dpnp.ones(N, dtype=request.param)
+    return a, b
+
+
+@pytest.fixture(params=list_of_int_dtypes)
+def input_arrays_for_and_or_xor(request):
+    # The size of input and out arrays to be used
+    N = 10
+    a = dpnp.arange(N, dtype=request.param)
+    b = dpnp.ones(N, dtype=request.param)
+    return a, b
+
+
 @pytest.mark.parametrize("ref_index", [0, 5])
-def test_fetch_add(input_arrays, ref_index):
+def test_fetch_add(input_arrays_for_add, ref_index):
     @dpex_exp.kernel
-    def atomic_ref_kernel(a, b, ref_index):
+    def atomic_add_kernel(a, b, ref_index):
         i = dpex.get_global_id(0)
         v = AtomicRef(b, index=ref_index)
         v.fetch_add(a[i])
 
-    a, b = input_arrays
+    a, b = input_arrays_for_add
 
-    dpex_exp.call_kernel(atomic_ref_kernel, dpex.Range(10), a, b, ref_index)
+    dpex_exp.call_kernel(atomic_add_kernel, dpex.Range(10), a, b, ref_index)
 
     # Verify that `a` was accumulated at b[ref_index]
     assert b[ref_index] == 10
 
 
+def test_fetch_add_diff_types():
+    """A negative test that verifies that a TypingError is raised if
```

Comment:
Same idea as the positive test cases should be applicable here to reduce the code duplication.